test/test_api/test_project_locks.py

```diff
@@ -0,0 +1,197 @@
+        })
+
+        projects = ProjectFactory.create_batch(5,
+            info={
+                'total': 150,
+                'task_presenter': 'foo',
+                'data_classification': dict(input_data="L4 - public", output_data="L4 - public"),
+                'product' : 'test_product',
+                'subproduct': 'test_subproduct2'
+        })
+        return project
+
+    def _create_dict_from_model(self, item):
+        # Dummy method for testing
+        return {}
+
+    def _sign_item(self, item):
+        # Dummy method for testing
+        pass
+
+    @with_context
+    def test_project_locks_user_not_logged_in(self):
+        """ Test should return 401 if the user is not logged in"""
+        project = self.setupProjects()
+        project_id = str(project.id)
+
+        res = self.app.get('/api/locks/' + project_id)
+        err = json.loads(res.data)
+        assert res.status_code == 401, err
+        assert err['status'] == 'failed', err
+        assert err['target'] == 'project', err
+        assert err['exception_cls'] == 'Unauthorized', err
+        assert err['action'] == 'GET', err
+
+    @with_context
+    def test_project_locks_user_worker(self):
+        """ Test API should return 401 if user is worker"""
+        admin = UserFactory.create(admin=True)
+        worker = UserFactory.create(admin=False, subadmin=False)
+
+        project = self.setupProjects()
+        project_id = str(project.id)
+
+        res = self.app.get('/api/locks?id=' + project_id + '&api_key=' + worker.api_key + '&all=1')
+        err = json.loads(res.data)
+        assert res.status_code == 401, err
+        assert err['status'] == 'failed', err
+        assert err['target'] == 'project', err
+        assert err['exception_cls'] == 'Unauthorized', err
+        assert err['action'] == 'GET', err
+
+    @with_context
+    def test_project_locks_user_subadmin(self):
+        """ Test API should work if user is subadmin"""
+        admin = UserFactory.create(admin=True)
+        subadmin = UserFactory.create(admin=False, subadmin=True)
+
+        project = self.setupProjects()
+        project_id = str(project.id)
+
+        res = self.app.get('/api/locks?id=' + project_id + '&api_key=' + subadmin.api_key + '&all=1')
+        data = json.loads(res.data)
+        assert res.status_code == 200, data
+        assert data[0]['short_name'] == 'test-app1', data
+
+    @with_context
+    def test_project_locks_user_admin(self):
+        """ Test API should work if user is admin"""
+        admin = UserFactory.create(admin=True)
+
+        project = self.setupProjects()
+        project_id = str(project.id)
+
+        res = self.app.get('/api/locks?id=' + project_id + '&api_key=' + admin.api_key + '&all=1')
+        data = json.loads(res.data)
+        assert res.status_code == 200, data
+        assert data[0]['short_name'] == 'test-app1', data
+
+    @with_context
+    def test_project_locks_get_by_id(self):
```

Comment:
This test seems redundant because the code is essentially the same as the one before it.

Reply:
The test specifically checks the return value of locks to be `[]`.

Reply:
This test specifically checks for the response data of the locks payload.

```
assert data[0]['locks'] == [], data
```